access/validator.go

```diff
@@ -396,6 +415,16 @@ func (v *TransactionValidator) checkSufficientBalanceToPayForTransaction(ctx con
 		return fmt.Errorf("could not fetch block header: %w", err)
 	}
 
+	indexedHeight, err := v.blocks.IndexedHeight()
+	if err != nil {
+		return fmt.Errorf("could not get indexed height: %w", err)
+	}
+
+	sealedHeight := header.Height
```

Comment:
can you add a comment here explaining why this is needed
```suggestion
	sealedHeight := header.Height
```